pkg/controllers/plugin/helm_chart_test_controller.go

```diff
@@ -96,7 +99,22 @@ func (r *HelmChartTestReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 		}
 	}()
 
-	clusterAccessReadyCondition, restClientGetter := initClientGetter(ctx, r.Client, r.kubeClientOpts, *plugin)
+	var restClientGetter genericclioptions.RESTClientGetter
+	clusterAccessReadyCondition := *pluginStatus.GetConditionByType(greenhousev1alpha1.ClusterAccessReadyCondition)
+
+	// Check if the plugin is ready to be reconciled.
```

Comment:
This should be done by the `initClientGetter`:
https://github.com/cloudoperators/greenhouse/blob/main/pkg/controllers/plugin/util.go#L77-L89

Reply:
I need this condition initialized before `initClientGetter` as I need to set condition if it fails getting `Cluster` resource which is passed to `initClientGetter` 

It was an attempt to avoid code duplication and fetching `Cluster` resource multiple times